src/js/common/schemaform/FormApp.jsx

```diff
@@ -102,6 +124,32 @@ class FormApp extends React.Component {
     return message;
   }
 
+  redirectOrLoad(props) {
+    // Stop a user that's been redirected to be redirected again after logging in
+    this.shouldRedirectOrLoad = false;
+
+    const firstPagePath = props.routes[props.routes.length - 1].pageList[0].path;
+
+    // If we're logged in and have a saved / pre-filled form, load that
+    if (props.isLoggedIn) {
+      const currentForm = props.formConfig.formId;
+      const isSaved = props.savedForms.some((savedForm) => savedForm.form === currentForm);
+      const isPrefill = props.prefillsAvailable.includes(currentForm);
+      const saveEnabled = !this.props.formConfig.disableSave;
+      if (saveEnabled && (isSaved || isPrefill)) {
```

Comment:
I added the SiP check in here so we still redirect back to the beginning if the form doesn't have SiP enabled.